blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -57,7 +58,8 @@ sealed abstract class BlazeClientBuilder[F[_]] private (
     val executionContext: ExecutionContext,
     val scheduler: Resource[F, TickWheelExecutor],
     val asynchronousChannelGroup: Option[AsynchronousChannelGroup],
-    val channelOptions: ChannelOptions
+    val channelOptions: ChannelOptions,
+    val customDnsResolver: Option[RequestKey => Either[Throwable, InetSocketAddress]]
```

Comment:
Not sure this needs to be an `Option`: could we just have a default based on the `getAddress` that's used today?

Reply:
Done. Also moved  `getAddress` to `BlazeClientBuilder`.